NGTrackForce/src/app/components/batch-list/batch-list.component.ts

```diff
@@ -205,133 +184,85 @@ export class BatchListComponent implements OnInit {
+				this.updateBatches();
   }
 
   // Logans new update batches method
+	// Refactored by Mussab
   public updateBatches()
   {
     const user = this.authService.getUser();
-    if (this.authService.getUserRole() === 2) {
-      // filter out batches that don't have an associated trainer
-      this.filteredBatches = this.batches.filter(
-        batch => {
-          if (batch.trainer.firstName !== this.authService.getTrainer().firstName) {
-            return false;
-          }
-          if (batch.coTrainer) {
-            if (!batch.coTrainer.includes(this.authService.getTrainer())) {
-              return false;
-            }
-          }
-          const dateStartDate = new Date(this.startDate);
-          const dateEndDate = new Date(this.endDate);
-          const longStartDate = dateStartDate.getTime();
-          const longEndDate = dateEndDate.getTime();
-
-          if (batch.startDate && batch.endDate) {
-            return batch.startDate > longStartDate && batch.endDate < longEndDate;
-          }
-          else {
-            return false;
-          }
+		this.dataReady = false;
+    this.filteredBatches = this.batches.filter(
+      batch => {
+				if (this.authService.getUserRole() === 2) {
+			     // filter out batches that don't have an association with the trainer
+					let trainer = batch.trainer.id !== this.authService.getTrainer().id;
+	        let coTrainer = batch.coTrainer && !batch.coTrainer.includes(this.authService.getTrainer());
+	        if (trainer && (coTrainer == undefined || coTrainer)) {
+	          return false;
+	        }
+				}
+        const dateStartDate = new Date(this.startDate);
+        const dateEndDate = new Date(this.endDate);
+        const longStartDate = dateStartDate.getTime();
+        const longEndDate = dateEndDate.getTime();
+
+        if (batch.startDate && batch.endDate) {
+          return batch.startDate >= longStartDate && batch.endDate <= longEndDate;
         }
-      );
-      this.updateCountPerCurriculum();
-      this.dataReady = true;
-    }
-    else{
-      this.dataReady = false;
-      this.filteredBatches = this.batches.filter(
-        batch => {
-          const dateStartDate = new Date(this.startDate);
-          const dateEndDate = new Date(this.endDate);
-          const longStartDate = dateStartDate.getTime();
-          const longEndDate = dateEndDate.getTime();
-
-          if (batch.startDate && batch.endDate) {
-            return batch.startDate > longStartDate && batch.endDate < longEndDate;
-          }
-          else {
-            return false;
-          }
+        else {
+          return false;
         }
-      );
-      this.updateCountPerCurriculum();
-      this.dataReady = true;
-    }
+      }
+    );
+
+		// Commented by Mussab cause issue #588 states the the chart
+		// be for all batches
+    // this.updateCountPerCurriculum();
+    this.dataReady = true;
   }
 
   //Steve L
+	// Refactored by Mussab
```

Comment:
Is this method ever used?

Reply:
Yes whenever you change the dates in the filter